runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java

```diff
@@ -881,7 +1005,81 @@ public void flatMap(
                   in.getValue().getKey(),
                   in.withValue(in.getValue().getValue()));
 
-          in.withValue(workItem);
+          out.collect(in.withValue(workItem));
```

Comment:
Wow, this looks like a fix for a serious bug - is it part of the current PR and does the current PR, directly or indirectly, add test coverage for it?